### PR TITLE
Fixed DebuggerDisplay in DefaultTagHelperContent

### PR DESCRIPTION
**Pull request by ajaybhargavb**
- Also refactored it in few other places

#1580 

@rynowak @NTaylorMullen

### CODE REVIEW

File: src/Microsoft.AspNetCore.Razor.Language/AllowedChildTagDescriptor.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 namespace Microsoft.AspNetCore.Razor.Language
 {
-    [DebuggerDisplay("{DisplayName,nq}")]
+    [DebuggerDisplay("{" + nameof(DisplayName) + ",nq}")]
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Language/AllowedChildTagDescriptor.cs]: This was better before 👎

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Language/AllowedChildTagDescriptor.cs]: It **looked** better before. But as @pranavkm [said](https://github.com/aspnet/Razor/issues/1580#issue-245158823),  this prevents bugs like these from happening again.

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Razor.Language/AllowedChildTagDescriptor.cs]: Strongly typed > Stringly typed

Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Language/AllowedChildTagDescriptor.cs]: After some discussion about this, we should make the following changes:
1. Override ToString where we have a display name property
2. Make a real `DebuggerToString()` for the places that don't use a display name property

**Comment by ajaybhargavb** [comment]
🆙 📅

**Pull request merged**